proxy/src/config.rs

```diff
@@ -1,16 +1,17 @@
 use crate::auth;
-use anyhow::{ensure, Context};
-use std::sync::Arc;
+use anyhow::{bail, ensure, Context};
+use std::{str::FromStr, sync::Arc, time::Duration};
 
 pub struct ProxyConfig {
     pub tls_config: Option<TlsConfig>,
     pub auth_backend: auth::BackendType<'static, ()>,
-    pub metric_collection_config: Option<MetricCollectionConfig>,
+    pub metric_collection: Option<MetricCollectionConfig>,
+    pub api_caches: auth::caches::ApiCaches,
```

Comment:
'api_caches' seems misplaced here. It's not configuration, it's the actual cache. Maybe rename this to ProxyGlobal or something: